packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -405,6 +405,28 @@ export default class ScriptTransformer {
     return result;
   }
 
+  transformJson(
+    filename: Config.Path,
+    options: Pick<Options, 'isInternalModule' | 'isCoreModule'> | undefined,
```

Comment:
it should take `Options`, not a subset of it

Reply:
I felt like it as needed to provide the full options, since not all properties were used.

In my latest commit, I changed to provide the full options object, and added a private helper method to create the full object from the `InternalModuleOptions`, in order to reduce code duplication.